tdrs-backend/tdpservice/data_files/util.py

```diff
@@ -3,54 +3,88 @@
 from io import BytesIO
 import xlsxwriter
 import calendar
-from tdpservice.parsers.models import ParserErrorCategoryChoices
+from django.conf import settings
+from django.core.paginator import Paginator
+from django.db import models
+from django.db.models import Count, Q
+from django.utils.translation import gettext_lazy as _
 
 
-def get_xls_serialized_file(data):
-    """Return xls file created from the error."""
+class ParserErrorCategoryChoices(models.TextChoices):
+    """Enum of ParserError error_type."""
+
+    PRE_CHECK = "1", _("File pre-check")
+    FIELD_VALUE = "2", _("Record value invalid")
+    VALUE_CONSISTENCY = "3", _("Record value consistency")
+    CASE_CONSISTENCY = "4", _("Case consistency")
+    SECTION_CONSISTENCY = "5", _("Section consistency")
+    HISTORICAL_CONSISTENCY = "6", _("Historical consistency")
+
+
+def get_prioritized_queryset(parser_errors):
+    """Generate a prioritized queryset of ParserErrors."""
+    PRIORITIZED_CAT2 = (
+        ("FAMILY_AFFILIATION", "CITIZENSHIP_STATUS", "CLOSURE_REASON"),
+    )
+    PRIORITIZED_CAT3 = (
+        ("FAMILY_AFFILIATION", "SSN"),
+        ("FAMILY_AFFILIATION", "CITIZENSHIP_STATUS"),
+        ("AMT_FOOD_STAMP_ASSISTANCE", "AMT_SUB_CC", "CASH_AMOUNT", "CC_AMOUNT", "TRANSP_AMOUNT"),
```

Comment:
interesting how this seems to also catch the tribal TANF version of this cat 3 error, which includes the following fields (some of which arent referenced here):

`TANF Child Care Care Amount,Amount of Subsidized Child Care,Cash Amount,Other Amount,Transportation Amount,Amount of Food Stamp Assistance,Transitional Services Amount`

![Screenshot 2024-12-05 094543](https://github.com/user-attachments/assets/329e113e-0135-4d9c-b982-83e595777144)

Reply:
Yes, the reason that we get both covered is because of the type of query I used. It checks that error contains at least all of the fields listed, but could contain more. That is why we get both since they share the same fields, Tribal just has a few extra.